RecoLocalCalo/HcalRecAlgos/src/HFSimpleTimeCheck.cc

```diff
@@ -0,0 +1,133 @@
+#include <cstring>
+#include <climits>
+
+#include "RecoLocalCalo/HcalRecAlgos/interface/HFSimpleTimeCheck.h"
+#include "RecoLocalCalo/HcalRecAlgos/interface/HFRecHitAuxSetter.h"
+
+HFSimpleTimeCheck::HFSimpleTimeCheck(const std::pair<float,float> tlimits[2],
+                                     const float energyWeights[2*HFAnodeStatus::N_POSSIBLE_STATES-1][2],
```

Comment:
Sorry... are we talking about the same thing? Arrays are never
passed by value. The difference between passing by reference
and the "C-style" argument (as in this code) is that in the "C-style"
usage the array variable immediately decays into a pointer, while
the reference allows for several additional operations. Here is
a useful article explaining the difference:

http://www.informit.com/guides/content.aspx?g=cplusplus&seqNum=241

On 06/17/2016 04:27 AM, David Lange wrote:
>
> In RecoLocalCalo/HcalRecAlgos/src/HFSimpleTimeCheck.cc <https://github.com/cms-sw/cmssw/pull/14838#discussion_r67482942>:
>
> > @@ -0,0 +1,133 @@
> > +#include <cstring>
> > +#include <climits>
> > +
> > +#include "RecoLocalCalo/HcalRecAlgos/interface/HFSimpleTimeCheck.h"
> > +#include "RecoLocalCalo/HcalRecAlgos/interface/HFRecHitAuxSetter.h"
> > +
> > +HFSimpleTimeCheck::HFSimpleTimeCheck(const std::pair<float,float> tlimits[2],
> > +                                     const float energyWeights[2*HFAnodeStatus::N_POSSIBLE_STATES-1][2],
> On Jun 17, 2016, at 11:22 AM, Igor Volobouev ***@***.***> wrote: In RecoLocalCalo/HcalRecAlgos/src/HFSimpleTimeCheck.cc: > @@ -0,0 +1,133 @@ > +#include <cstring> > +#include <climits> > + > 
> +#include "RecoLocalCalo/HcalRecAlgos/interface/HFSimpleTimeCheck.h" > +#include "RecoLocalCalo/HcalRecAlgos/interface/HFRecHitAuxSetter.h" > + > +HFSimpleTimeCheck::HFSimpleTimeCheck(const 
> std::pair<float,float> tlimits[2], > + const float energyWeights[2*HFAnodeStatus::N_POSSIBLE_STATES-1][2], Yes, but passing by reference brings no advantage for the simple use of arrays in this 
> code, and any code change now would mean another week-long delay for collecting signatures
> it would cause little delay as we don’t wait for signatures a second time around for simple changes. how small is this array?
> … <#>
> … — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or mute the thread.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/14838/files/94ddf556520df6348e6f7f6215e44341162a69a5#r67482942>, or mute the thread 
> <https://github.com/notifications/unsubscribe/AFS4-e1bA7lROboPKqtugmR6RxHfVs2Zks5qMmiCgaJpZM4IyfiD>.
>